src/Sylius/Behat/Page/Shop/Product/IndexPage.php

```diff
@@ -24,31 +24,31 @@ public function getRouteName(): string
 
     public function countProductsItems(): int
     {
-        $productsList = $this->getDocument()->find('css', '#products');
+        $productsList = $this->getDocument()->find('css', '[data-test-products]');
 
-        $products = $productsList->findAll('css', '.card');
+        $products = $productsList->findAll('css', '[data-test-product]');
 
         return count($products);
     }
 
     public function getFirstProductNameFromList(): string
     {
-        $productsList = $this->getDocument()->find('css', '#products');
+        $productsList = $this->getDocument()->find('css', '[data-test-products]');
 
-        return $productsList->find('css', '.card:first-child .content > a')->getText();
+        return $productsList->find('css', '[data-test-product]:first-child [data-test-product-content] > a')->getText();
     }
 
     public function getLastProductNameFromList(): string
     {
-        $productsList = $this->getDocument()->find('css', '#products');
+        $productsList = $this->getDocument()->find('css', '[data-test-products]');
 
-        return $productsList->find('css', '.card:last-child .content > a')->getText();
+        return $productsList->find('css', '[data-test-product]:last-child [data-test-product-content] > a')->getText();
     }
 
     public function search(string $name): void
     {
         $this->getDocument()->fillField('criteria_search_value', $name);
-        $this->getDocument()->pressButton('Search');
+        $this->getDocument()->find('css', '[data-test-search]')->getParent()->submit();
```

Comment:
Can you add this attribute on button and have :
```suggestion
        $this->getDocument()->find('css', '[data-test-search]')->submit();
```